### PR TITLE
TRITON-2011 remove dependency on node-lockfd

### PR DESCRIPTION
**Pull request by mgerdts**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,16 +13,17 @@
...
     "jsprim": "2.0.0",
     "kstat": "1.0.1",
     "lazy-property": "1.0.0",
-    "lockfd": "1.2.0",
     "lstream": "0.0.4",
     "mkdirp": "^0.3.4",
     "nodeunit": "0.9.1",
     "once": "1.3.0",
     "pty.js": "0.3.1",
+    "qlocker": "git+https://github.com/joyent/node-qlocker.git#ec3c5323e4070fa83b39508f7199efab4e6d055b",
```
Comment (twhiteman) [inline package.json]: Nit - can qlocker be published in npm - as mostly we prefer npm over git urls.

BTW - curious on why we don't use an existing node lockfile module, e.g.
https://www.npmjs.com/package/lockfile

File: package.json [hunk 2]
```diff
@@ -13,16 +13,17 @@
     "cueball": "2.7.1",
     "digest-stream": "0.2.2",
     "forkexec": "1.1.0",
+    "fs-ext": "1.3.0",
```
[the remark this answers is not in the document]
Comment (mgerdts) [reply package.json]: oops, left over from before I slit qlocker into its own repo.

File: package.json (hunk 1 above)
Comment (mgerdts) [reply package.json]: > Nit

ok, published

> BTW

That only handles inter-process locks.  We need inter-process and intra-process locks.

```
$ cat test
#! /usr/bin/env node

lf = require('lockfile')

lf.lock('/tmp/lock', function() {
    lf.lock('/tmp/lock', function() {
        console.log('got it twice')
    })
})

$ ./test
got it twice
```

Comment (twhiteman) [reply package.json]: Thanks - that helps :)

**Pull request merged**